flake8_docstrings.py

```diff
@@ -103,21 +140,32 @@ def parse_options(cls, options):
             if options.ignore_decorators
             else None
         )
+        if supports_property_decorators:
+            cls.property_decorators = options.property_decorators
+        if supports_ignore_self_only_init:
+            cls.ignore_self_only_init = options.ignore_self_only_init
 
     def _call_check_source(self):
-        try:
-            return self.checker.check_source(
-                self.source,
-                self.filename,
-                ignore_decorators=self.ignore_decorators,
-                ignore_inline_noqa=True,
-            )
-        except TypeError:  # for versions of pydocstyle 5.1.1 and below
-            return self.checker.check_source(
-                self.source,
-                self.filename,
-                ignore_decorators=self.ignore_decorators,
+        check_source_kwargs = {}
+        if supports_ignore_inline_noqa:
+            check_source_kwargs["ignore_inline_noqa"] = True
+        if supports_property_decorators:
+            check_source_kwargs["property_decorators"] = (
+                set(self.property_decorators.split(","))
+                if self.property_decorators
+                else None
             )
+        if supports_ignore_self_only_init:
+            check_source_kwargs[
+                "ignore_self_only_init"
+            ] = self.ignore_self_only_init
+
+        return self.checker.check_source(
+            self.source,
+            self.filename,
+            ignore_decorators=self.ignore_decorators,
+            **check_source_kwargs,
```

Comment:
Thank you for using kwargs here. pydocstyle has had a bunch of issues with other integrations using positional args